### PR TITLE
Order component fix

### PR DESCRIPTION
**Pull request by jul4ajka**
New functionality for order component

### CODE REVIEW

File: src/pages/order-item/order-item.js [hunk 1]
```diff
@@ -28,24 +28,49 @@ const OrderItem = ({ id }) => {
...
+      status: order.status,
+      user: order.user,
+      delivery: order.delivery,
+      items,
+      paymentMethod: order.paymentMethod,
+      userComment: order.userComment,
+      isPaid: order.isPaid,
+      paymentStatus: order.paymentStatus
+    };
+    console.log(newOrder);
```
Comment (AndriyLymych) [inline src/pages/order-item/order-item.js:62]: remove console

File: src/pages/order-item/tabs/products.js [hunk 2]
```diff
@@ -1,51 +1,49 @@
...
+        name={item.product.name[0].value}
+        quantity={item.quantity}
+        price={`${item.product.basePrice[0].value * item.quantity }₴`}
+        showAvatar={false}
+        deleteHandler={() => deleteItemHendler(index)}
+        editHandler={() => setSelectedItem(item)}
+      />
+    ));
 
+  console.log(selectedItem);
```
Comment (AndriyLymych) [inline src/pages/order-item/tabs/products.js:46]: remove console

File: src/pages/order-item/order-item.js [hunk 3]
```diff
@@ -28,24 +28,49 @@ const OrderItem = ({ id }) => {
...
 
   const handleFormSubmit = (order) => {
+    const items = order.items.map((item) => ({
+      product: item?.product._id,
+      quantity: item.quantity,
+      isFromConstructor: !item.product._id,
+      options: {
+        size: item.options.size._id,
+        sidePocket: item.options.sidePocket
+      }
```
Comment (AndriyLymych) [inline src/pages/order-item/order-item.js:50]: create in utils file order (or use already created) and put this func into utils

File: src/pages/order-item/order-item.js [hunk 4]
```diff
@@ -28,24 +28,49 @@ const OrderItem = ({ id }) => {
...
+    const newOrder = {
+      status: order.status,
+      user: order.user,
+      delivery: order.delivery,
+      items,
+      paymentMethod: order.paymentMethod,
+      userComment: order.userComment,
+      isPaid: order.isPaid,
+      paymentStatus: order.paymentStatus
+    };
```
Comment (AndriyLymych) [inline src/pages/order-item/order-item.js:61]: same

File: src/pages/order-item/tabs/delivery.js [hunk 5]
```diff
@@ -1,16 +1,31 @@
...
+  const { delivery } = data;
+  const { sentOn, sentBy, byCourier, courierOffice, cost } = delivery;
+
+  const address = {
+    city: delivery.city,
+    street: delivery.street,
+    house: delivery.house,
+    flat: delivery.flat
+  };
+
```
Comment (AndriyLymych) [inline src/pages/order-item/tabs/delivery.js:28]: put into utils same

File: src/pages/order-item/tabs/delivery.js [hunk 6]
```diff
@@ -20,31 +35,66 @@ const Delivery = ({data,handleChange}) => {
...
-      <div className={classes.idContainer}>
-        <label htmlFor="byCourier">{byCourierLabel}</label>
-        <Checkbox id='byCourier' checked={byCourier} name='delivery.byCourier' onChange={handleChange}/>
-      </div>
       <TextField
-        label={invoiceNumberLabel}
-        name='delivery.invoiceNumber'
-        value={invoiceNumber}
+        label={courierOfficeNameLabel}
+        name='delivery.office'
```
Comment (AndriyLymych) [inline src/pages/order-item/tabs/delivery.js:40]: no magic strings

File: src/pages/order-item/tabs/delivery.js [hunk 7]
```diff
@@ -20,31 +35,66 @@ const Delivery = ({data,handleChange}) => {
...
+        value={courierOffice}
         onChange={handleChange}
         variant='outlined'
```
Comment (AndriyLymych) [inline src/pages/order-item/tabs/delivery.js:43]: same

File: src/pages/order-item/tabs/delivery.js [hunk 8]
```diff
@@ -20,31 +35,66 @@ const Delivery = ({data,handleChange}) => {
...
       />
       <div className={classes.dateContainer}>
-        <p>{warehouseNumberLabel} {courierOffice}</p>
-        <p>{sentAtLabel} {moment.unix(sentOn/1000).format(' HH:mm DD/MM/YYYY ')}</p>
+        {sentOn ? (
+          <p>
+            {sentAtLabel}{' '}
+            {moment.unix(sentOn / 1000).format(' HH:mm DD/MM/YYYY ')}
```
Comment (AndriyLymych) [inline src/pages/order-item/tabs/delivery.js:49]: format in magic string

File: src/pages/order-item/tabs/delivery.js [hunk 9]
```diff
@@ -20,31 +35,66 @@ const Delivery = ({data,handleChange}) => {
...
-        name='delivery.cost[0].value'
-        value={cost[0].value}
-        onChange={handleChange}
-        variant='outlined'
-      />
-      {byCourier && <DeliveryDetails address={address} handleChange={handleChange}/>}
+      {delivery?.cost[0]?.value ? (
+        <TextField
+          label={deliveryCostLabel}
+          name='delivery.cost[0].value'
```
Comment (AndriyLymych) [inline src/pages/order-item/tabs/delivery.js:56]: same

File: src/pages/order-item/tabs/delivery.js [hunk 10]
```diff
@@ -20,31 +35,66 @@ const Delivery = ({data,handleChange}) => {
...
+          value={cost[0].value}
+          onChange={handleChange}
+          variant='outlined'
```
Comment (AndriyLymych) [inline src/pages/order-item/tabs/delivery.js:59]: same

File: src/pages/order-item/tabs/delivery.js [hunk 11]
```diff
@@ -20,31 +35,66 @@ const Delivery = ({data,handleChange}) => {
...
+        />
+      ) : null}
+      <div className={classes.idContainer}>
+        <label htmlFor='byCourier'>{byCourierLabel}</label>
+        <Checkbox
+          id='byCourier'
+          checked={byCourier}
+          name='delivery.byCourier'
```
Comment (AndriyLymych) [inline src/pages/order-item/tabs/delivery.js:67]: same

File: src/pages/order-item/tabs/general.js [hunk 12]
```diff
@@ -1,67 +1,112 @@
...
-        <Select fullWidth id='status' name='status' value={status} onChange={handleChange} variant='outlined' defaultValue={statusOptions[0].value}>
+        <label htmlFor='status'>{deliveryStatusLabel}</label>
+        <Select
+          fullWidth
+          id='status'
+          name='status'
+          value={status}
+          onChange={handleChange}
+          variant='outlined'
+          defaultValue={statusOptions[0].value}
```
Comment (AndriyLymych) [inline src/pages/order-item/tabs/general.js:53]: name,variant magic strings

File: src/pages/order-item/tabs/general.js [hunk 13]
```diff
@@ -1,67 +1,112 @@
...
-        <label htmlFor="paymentMethod">{paymentMethodLabel}</label>
-        <Select fullWidth id='paymentMethod' name='paymentMethod' value={paymentMethod} onChange={handleChange} variant='outlined' defaultValue={paymentOptions[0].value}>
+        <label htmlFor='paymentStatus'>{paymentStatusLabel}</label>
+        <Select
+          fullWidth
+          id='paymentStatus'
+          name='paymentStatus'
+          value={paymentStatus}
+          onChange={handleChange}
+          variant='outlined'
```
Comment (AndriyLymych) [inline src/pages/order-item/tabs/general.js:66]: same

File: src/pages/order-item/tabs/general.js [hunk 14]
```diff
@@ -1,67 +1,112 @@
...
+      <div>
+        <label htmlFor='paymentMethod'>{paymentMethodLabel}</label>
+        <Select
+          fullWidth
+          id='paymentMethod'
+          name='paymentMethod'
+          value={paymentMethod}
+          onChange={handleChange}
+          variant='outlined'
+          defaultValue={paymentOptions[0].value}
```
Comment (AndriyLymych) [inline src/pages/order-item/tabs/general.js:81]: same

File: src/pages/order-item/tabs/general.js [hunk 15]
```diff
@@ -1,67 +1,112 @@
...
+        >
           {paymentOptionsElements}
         </Select>
       </div>
       <div className={classes.isPaid}>
-        <label htmlFor="paymentMethod">{isPaidLabel}</label>
-        <Checkbox checked={isPaid} name='isPaid' onChange={handleChange}/>
+        <label htmlFor='paymentMethod'>{isPaidLabel}</label>
```
Comment (AndriyLymych) [inline src/pages/order-item/tabs/general.js:87]: same

File: src/pages/order-item/tabs/general.js [hunk 16]
```diff
@@ -1,67 +1,112 @@
...
+        <Checkbox checked={isPaid} name='isPaid' onChange={handleChange} />
```
Comment (AndriyLymych) [inline src/pages/order-item/tabs/general.js:88]: same

File: src/pages/order-item/tabs/products.js [hunk 17]
```diff
@@ -1,51 +1,49 @@
...
-        <label htmlFor={label}><b>{label}:</b></label>
-        <ul id={label} className={classes.renderList}>
-          {renderItems.map(item => (
-            <li key={item[0].value}>{item[0].value}</li>
-          ))}
-        </ul>
-      </div>
+  const deleteItemHendler = (indexItem) => {
+    setFieldValue(
+      'items',
```
Comment (AndriyLymych) [inline src/pages/order-item/tabs/products.js:26]: same

File: src/pages/order-item/tabs/products.js [hunk 18]
```diff
@@ -54,32 +52,34 @@ const Products = ({data,setFieldValue}) => {
...
-          {renderList(colors,colorsLabel)}
-          {renderList(sizes,sizeLabel)}
+          <h2 className={classes.productHeading}>
+            {selectedItem.product && selectedItem.product.name[0].value}
+          </h2>
+          <div>Кількість: {selectedItem.quantity}</div>
+          <br />
+          <div>Розмір: {selectedItem?.options?.size.name}</div>
+          <br />
+          <div>Ціна: {selectedItem?.product?.basePrice[0].value}</div>
```
Comment (AndriyLymych) [inline src/pages/order-item/tabs/products.js:66]: same

File: src/pages/orders/orders-page.js [hunk 19]
```diff
@@ -51,28 +57,34 @@ const OrdersPage = () => {
...
+  };
 
   const orderItems =
     orders &&
     orders.map((order) => (
       <TableContainerRow
         key={order._id}
-        orderId={order._id}
+        orderId={order.orderNumber}
         date={moment.unix(order.dateOfCreation / 1000).format(' DD.MM.YYYY ')}
```
Comment (AndriyLymych) [inline src/pages/orders/orders-page.js:76]: same

File: src/pages/orders/orders-page.js [hunk 20]
```diff
@@ -87,6 +99,15 @@ const OrdersPage = () => {
           <Select value={status} onChange={(e) => setStatus(e.target.value)}>
             {statusList}
           </Select>
+          <Button
+            id='add-order'
+            component={Link}
+            to={pathToOrdersAddPage}
+            variant='contained'
```
Comment (AndriyLymych) [inline src/pages/orders/orders-page.js:106]: same

File: src/redux/orders/orders.operations.js [hunk 21]
```diff
@@ -3,244 +3,180 @@ import { getItems, setItems, client } from '../../utils/client';
...
+					paymentMethod
+					paymentStatus
+					isPaid
+				}
+				...on Error {
+					statusCode
+					message
+				}
+			}
+		}
```
Comment (AndriyLymych) [inline src/redux/orders/orders.operations.js:105]: prettify this

File: src/redux/orders/orders.operations.js [hunk 22]
```diff
@@ -3,244 +3,180 @@ import { getItems, setItems, client } from '../../utils/client';
...
+				paymentMethod
+				paymentStatus
+				isPaid
+			}
+			...on Error {
+				statusCode
+				message
+			}
+		}
+	}
```
Comment (AndriyLymych) [inline src/redux/orders/orders.operations.js:177]: same

**Comment by AndriyLymych** [comment]
image
 fix tests

**Review by AndriyLymych — APPROVED** [review]

**Review by Kiyooooko — APPROVED** [review]

**Review by AndriyLymych — APPROVED** [review]

**Pull request merged**